src/common/hooks/useGenericState.ts

```diff
@@ -0,0 +1,14 @@
+import React, { useState } from 'react'
+
+export default function useGenericState<T>(state: T): [T, (newState: Partial<T>) => void] {
```

Comment:
Avoid passing inline types

Reply:
I don't think we need this wrapper - it's unnecessary wrapping around useState
If the goal is to try and create a class component like `setState` functionality, then we shouldn't be doing that in the first place
React hooks were built to have modularized immutability
If you wanna expose a common state with a setter - use `useReducer` instead